hibernate-core/src/test/java/org/hibernate/dialect/resolver/DialectFactoryTest.java

```diff
@@ -136,9 +142,20 @@ public void testPreregisteredDialects() {
 		testDetermination( "H2", H2Dialect.class, resolver );
 		testDetermination( "MySQL", MySQLDialect.class, resolver );
 		testDetermination( "MySQL", 5, 0, MySQL5Dialect.class, resolver );
+		testDetermination( "MySQL", 5, 5, MySQL55Dialect.class, resolver );
+		testDetermination( "MySQL", 5, 6, MySQL55Dialect.class, resolver );
+		testDetermination( "MySQL", 5, 7, MySQL57Dialect.class, resolver );
+		testDetermination( "MySQL", 8, 0, MySQL57Dialect.class, resolver );
 		testDetermination( "PostgreSQL", PostgreSQL81Dialect.class, resolver );
 		testDetermination( "PostgreSQL", 8, 2, PostgreSQL82Dialect.class, resolver );
+		testDetermination( "PostgreSQL", 8, 3, PostgreSQL82Dialect.class, resolver );
 		testDetermination( "PostgreSQL", 9, 0, PostgreSQL9Dialect.class, resolver );
+		testDetermination( "PostgreSQL", 9, 1, PostgreSQL9Dialect.class, resolver );
+		testDetermination( "PostgreSQL", 9, 2, PostgreSQL92Dialect.class, resolver );
+		testDetermination( "PostgreSQL", 9, 3, PostgreSQL92Dialect.class, resolver );
+		testDetermination( "PostgreSQL", 9, 4, PostgreSQL94Dialect.class, resolver );
+		testDetermination( "PostgreSQL", 9, 5, PostgreSQL95Dialect.class, resolver );
+		testDetermination( "PostgreSQL", 10, 0, PostgreSQL95Dialect.class, resolver );
```

Comment:
This change seems good to me. I'm a little unclear on the goal of this test class -- is it trying to check the dialect for every major.minor version, or just a few? For example, we're not checking versions 8.1, 9.6, or 10.1 of PostgreSQL. I know that new versions are released periodically, which would mean coming back to this test to add a test case for that version, so perhaps we're just trying to check some sample?

Reply:
The goal of this test case is to prove that versions are picked correctly. I haven't listed every possible version, but, if 10.0 is resolved, 10.1 should be resolved as well.

Reply:
LGTM then!